### PR TITLE
Add a custom file offline store implementation

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,54 @@
-# feast-custom-offline-store-demo
\ No newline at end of file
+# Feast Custom Offline Store
+[![test-offline-store](https://github.com/feast-dev/feast-custom-offline-store-demo/actions/workflows/test_custom_offline_store.yml/badge.svg?branch=main)](https://github.com/feast-dev/feast-custom-offline-store-demo/actions/workflows/test_custom_offline_store.yml)
+
+### Overview
+
+This repository demonstrates how developers can create their own custom `offline store`s for Feast.
+Custom offline stores allow users to use any underlying data store to as the source of truth for features which get materialized into the online feature store. Features can be retrieved from the offline store for model training.
```
Comment (woop) [inline README.md]: > Custom offline stores allow users to use any underlying data store to as the source of truth for features which get materialized into the online feature store. Features can be retrieved from the offline store for model training.

This sentence is a bit confusing. What does it mean to have the offline store as the source of truth? I thought the source was the source of truth?

Loading data into the online store and building training datasets should probably be connected as a single sentence, it's a bit jarring to read right now.

Comment (woop) [reply README.md]:
```suggestion
Custom offline stores allow users to use any underlying data store as the source of truth for features which get materialized into the online feature store. Features can be retrieved from the offline store for model training. 
```

File: README.md [hunk 2]
```diff
@@ -1 +1,54 @@
...
+
+### Overview
+
+This repository demonstrates how developers can create their own custom `offline store`s for Feast.
+Custom offline stores allow users to use any underlying data store to as the source of truth for features which get materialized into the online feature store. Features can be retrieved from the offline store for model training. 
+
+
+### Why create a custom offline store?
+
+Feast uses an offline store as the source of truth for features. These features can be retrieved from the offline store for model training. Typically, scalable Data Warehouses are used for this purpose.
```
Comment (woop) [inline README.md]: Should `Data Warehouses` be capitalized?

File: README.md [hunk 3]
```diff
@@ -1 +1,54 @@
...
+ 
+Feast also materializes features from offline stores to an online store for low-latency lookup at model inference time. 
+
+Feast comes with some offline stores built in, e.g, Parquet file, Redshift and Bigquery. However, users can develop their own offline stores by creating a class that implements the contract in the [OfflineStore class](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/offline_stores/offline_store.py#L41).
```
Comment (woop) [inline README.md]: I think we should use a permalink, just in case.

File: test_custom_offline_store.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+from feast import FeatureStore
+
+from feature_repo.repo import driver, driver_hourly_stats_view
+
+
+def test_end_to_end():
+    fs = FeatureStore("feature_repo/")
+
+    try:
```
Comment (woop) [inline test_custom_offline_store.py:13]: but what happens within this try/finally if there is an exception? Does it raise the error?

File: .envrc [hunk 5]
```diff
@@ -0,0 +1 @@
+layout_python3
```
Comment (woop) [inline .envrc]: what is this?

Comment (achals) [reply .envrc]: Oh I use [direnv](https://direnv.net/) for manage virtualenvs and this configures that. I'll gitignore it

File: README.md (hunk 1 above)
Comment (achals) [reply README.md]: I can update this to be clearer. I guess it's more accurate to the say that the data source if the source of truth - and I'll move the sentences a bit.

File: README.md (hunk 2 above)
Comment (achals) [reply README.md]: I can un-capitalize

File: test_custom_offline_store.py (hunk 4 above)
Comment (achals) [reply test_custom_offline_store.py:13]: Yes, it raises the exception. It would still fail the test but make an attempt to clean up the repo.

**Pull request merged**